overlord/snapstate/flags.go

```diff
@@ -50,3 +54,32 @@ func (f Flags) ForSnapSetup() Flags {
 	f.IgnoreValidation = false
 	return f
 }
+
+// EffectiveConfinement return the effective confinement for a given declared
+// confinement and flags like jailmode or devmode.
+func (f Flags) EffectiveConfinement(confinementType snap.ConfinementType) snap.ConfinementType {
+	switch confinementType {
+	case snap.ClassicConfinement:
+		// unchanged, flags don't impact snaps with classic confinement
+	case snap.DevmodeConfinement:
+		if f.JailMode {
+			// jailmode overrides devmode confinement declaration
```

Comment:
Comments which simply restate exactly what the logic does in general are a negative asset.

If the code isn't yet as readable as the comment, it can often be reorganized so that it is. For example:
```
if snapOpinion == snap.Devmode && flags.Jailmode {
        return snap.Strict
}
```

Note I dropped the `Confinement` suffixes as it feels like unnecessary noise given that those are strongly typed variables which can only be used within the proper context.